modules/os-eap-datasource/1.0/added/launch/datasource-common.sh

```diff
@@ -307,6 +310,17 @@ function generate_external_datasource() {
            </validation>"
   fi
 
+  if [ -n "$conn_properties" ]; then
+    properties_array=$(echo $conn_properties | tr ';' '\n' )
```

Comment:
The `echo` is not needed here. Pass the needed variable to the translate (`tr`) command directly. E.g.:
```
[iankko@localhost ~]$ export conn_properties="first;second;third"
[iankko@localhost ~]$ tr ';' '\n' <<< "${conn_properties}"
first
second
third
```

Reply:
:+1 , I see the benefit of your approach on reducing the sub-shells